File: GuardeSoftwareAPI/Dtos/Client/CreateClientDTO.cs
using System;
using GuardeSoftwareAPI.Dtos.Address;

namespace GuardeSoftwareAPI.Dtos.Client
{

    public class CreateClientDTO
    {
        //Client table
        public int? Id { get; set; }
        public decimal? PaymentIdentifier { get; set; }
        public string FirstName { get; set; } = string.Empty;
        public string LastName { get; set; } = string.Empty;
        public DateTime RegistrationDate { get; set; }
        public string? Notes { get; set; } = string.Empty;
        public string? Dni { get; set; } = string.Empty;
        public string? Cuit { get; set; } = string.Empty;
        public int? PreferredPaymentMethodId { get; set; }
        public string? IvaCondition { get; set; } = string.Empty;

        //Rental Table
        public DateTime StartDate { get; set; }
        // Maybe add End date??
        public decimal? ContractedM3 { get; set; }

        //rentalAmountHistory
        public decimal Amount { get; set; }
        //id of locker rental
        public List<int> LockerIds { get; set; } = [];
        //user id for activityLog
        public int UserID { get; set; }

        //Contact Info
        public List<string> Emails { get; set; } = [];
        public List<string> Phones { get; set; } = [];
        public CreateAddressDto AddressDto { get; set; } = new CreateAddressDto();
    }
}

File: GuardeSoftwareClient/src/app/core/dtos/client/CreateClientDTO.ts
import { CreateAddressDto } from "../address/CreateAddressDto";
import { SpaceRequestDto } from "../rentalSpaceRequest/SpaceRequestDto";

export interface CreateClientDTO{
    
    id?: number;
    paymentIdentifier: number;
    firstName: string;
    lastName: string;
    registrationDate: Date;
    notes: string;
    dni: string;
    cuit?: string;
    preferredPaymentMethodId: number;
    ivaCondition: string;
    billingType?: string;
    billingTypeId?: number;
    startDate: Date;
    amount: number;
    userID: number;

    //locker information
    spaceRequests: SpaceRequestDto[]; 
    lockerIds: number[];
    occupiedSpaces: number;
    contractedM3: number;

    //contact info
    emails: string[];
    phones: string[];
    addressDto: CreateAddressDto,

    // Legacy client fields
    prepaidMonths?: number;
    isLegacyClient?: boolean;
    legacyInitialAmount?: number;
    legacyNextIncreaseDate?: Date;
    isLegacy6MonthPromo?: boolean;


    // increaseFrequency: string;
    // increasePercentage: number;
}